### PR TITLE
Catch dbal conversion exception

### PR DESCRIPTION
**Pull request by norkunas**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #
| License       | MIT
| Doc PR        | #

Using [ramsey/uuid-doctrine](https://github.com/ramsey/uuid-doctrine) doctrine type for entities I've accidentally encountered `Doctrine\DBAL\Types\ConversionException` which was thrown in `ApiPlatform\Core\Bridge\Doctrine\Orm\Util\IdentifierManagerTrait` and the response was 500 instead of 400.
This is possible if user enters a different ID than uuid when fetching a resource. Rethrowing as `InvalidArgumentException` apip properly returns the 400 code.

In slack I didn't got any response from the core team so submitting here to get some feedback. If this patch will be considered to merge I'll try to add test.

### CODE REVIEW

File: src/Bridge/Doctrine/Orm/Util/IdentifierManagerTrait.php [hunk 1]
```diff
@@ -77,7 +79,11 @@ private function normalizeIdentifiers($id, ObjectManager $manager, string $resou
             $doctrineTypeName = $doctrineClassMetadata->getTypeOfField($propertyName);
 
             if ($isOrm && null !== $doctrineTypeName && DBALType::hasType($doctrineTypeName)) {
-                $identifier = DBALType::getType($doctrineTypeName)->convertToPHPValue($identifier, $platform);
+                try {
+                    $identifier = DBALType::getType($doctrineTypeName)->convertToPHPValue($identifier, $platform);
+                } catch (ConversionException $e) {
+                    throw new InvalidArgumentException(sprintf('Invalid identifier "%s" value provided.', $propertyName));
```
Comment (soyuka) [inline src/Bridge/Doctrine/Orm/Util/IdentifierManagerTrait.php]: We should keep the previous exception as done https://github.com/api-platform/core/pull/1304/files. Not sure about the use case though.

**Comment by dkarlovi** [comment]
#1309 is related.

**Comment by soyuka** [comment]
@dkarlovi :+1:  thanks for the reminder! I may try to take a shot at this, I'm sure it can improve the DX as well

**Comment by norkunas** [comment]
#1478 replaces this so closing then.

**Pull request closed without merging**